src/swap/types.ts

```diff
@@ -59,12 +62,116 @@ export interface V2SwapExecution {
   /** Can be `undefined` if the execution was successful, but there was an issue while
    * extracting the output asset data fron the transaction response */
   assetOut: AssetWithIdAndAmount | undefined;
-  pool: V2PoolInfo;
+  quote: SwapQuote;
   txnID: string;
   round: number;
 }
 
+export type GetSwapQuoteParams = {
+  assetIn: Pick<TinymanAnalyticsApiAsset, "id" | "decimals">;
+  assetOut: Pick<TinymanAnalyticsApiAsset, "id" | "decimals">;
+  pools: {info: V1PoolInfo | V2PoolInfo; reserves: PoolReserves}[];
+  amount: number | bigint;
+  type: SwapType;
+  network: SupportedNetwork;
+  /** If `true`, the function will also check the quotes that use swap route */
+  isSwapRouterEnabled?: boolean;
+};
+
 export interface SwapQuoteWithPool {
+  quote: DirectSwapQuote;
+  pool: V1PoolInfo | V2PoolInfo;
+}
+
+export type SwapQuote =
+  | {
+      quoteWithPool: SwapQuoteWithPool;
+      type: SwapQuoteType.Direct;
+    }
+  | (FetchSwapRouteQuotesResponse & {
```

Comment:
this would be better IMO:
```suggestion
  |  {
    routerResponse: FetchSwapRouteQuotesResponse
```

or, even better:

```
export type SwapQuote =
  | {
      type: SwapQuoteType.Direct;
      data: SwapQuoteWithPool;
    }
  | {
      type: SwapQuoteType.Router;
      data: FetchSwapRouteQuotesResponse;
    }
```

Would it also make sense to rename these: 
`SwapQuoteWithPool` -> `DirectSwapQuoteAndPool`
`FetchSwapRouteQuotesResponse` -> `SwapRouterResponse`

Also, I agree with Cem's comment about declaring the types in a particular order that would make it easier to read from top to bottom. The grouping could be something like:

```
common types, eg. SwapType

direct quote related types

swap router related types (they should be ordered too: eg. SwapRouterQuote should be defined earlier than SwapRoute)

quote helper util function related types

SwapQuote

Swap txn related types

Execution types
```